### ISSUE TITLE
Release 5.2

### ISSUE THREAD

**Issue by pooyan-dadvand** [KratosMultiphysics/Kratos#866]
The next release has been scheduled for the end of the November. So at the end of this month we will make the release branch from master to be used for interface refinement and bug fixing. 

It is very important to add your application to the linux and windows binaries in order to have them available and downloadable for the next 3 month for other people without need to compile the code. (This is useful for collaborating with users who only work with python) The code at the moment of making release branch should contain all features and for release time be as stable as possible. I would like to emphasis that the effort for creating a release increases considerably the quality of the application and improves its maintainability. So I would strongly recommend application developers to make such effort.  

I have changed the previous release [project](https://github.com/KratosMultiphysics/Kratos/projects/5) to new one maintaining the same structure and also the application which where included. 

I have also created a [milestones](https://github.com/KratosMultiphysics/Kratos/milestone/4) for better organization

## Steps to take
I would encourage all @KratosMultiphysics/team-maintainers and also developers to:
1. Revising the [project](https://github.com/KratosMultiphysics/Kratos/projects/5) and add their application if they want to join this release.
2. Revising their corresponding issues and assign them to the  [release 5.2 milestone](https://github.com/KratosMultiphysics/Kratos/milestone/4) 

I would kindly ask for all your collaboration during this release period.

**Update**: I have realized that an issue cannot be added to two milestones. So I have removed one milestone to keep the organization easier.

**Comment by jcotela** [comment]
Hi @pooyan-dadvand, since the FluidDynamicsApplication is already there, I'll just say that I'd like to include it again in Release 5.2. I'll create issues for the new features/corrections we decided we wanted for the next release and add them to the milestones ASAP.

Update: issues now added to the milestone. Feel free to add any new bug reports for the FluidDynamicsApplication to that milestone too.

**Comment by RiccardoRossi** [comment]
@KratosMultiphysics/structural-mechanics, and here i refer in particular to people @ TUM. Can u take a look at this? from my side i really would like the "local axis" for the beam to be included in this release. I already contacted (today) @KlausBSautter about this. However there are more things to be improved

**Comment by pooyan-dadvand** [comment]
Thanks @jcotela for organizing your application plan. I would encourage you to search for the reported issues and classifying them by adding or not to the release milestone. 

Meanwhile, I strongly encourage other @KratosMultiphysics/team-maintainers to do the same before the next Tuesday so we can go organizing better looking to the [milestone](https://github.com/KratosMultiphysics/Kratos/milestone/4)

**Comment by pooyan-dadvand** [comment]
# 9 days to making release branch

Dear @KratosMultiphysics/team-maintainers This is a reminder that all features to be included in the release should be merged to master before end of the month. Still the input from @KratosMultiphysics/dem, @KratosMultiphysics/structural-mechanics and FSI are missing.

**Comment by pooyan-dadvand** [comment]
# 8 days to making release branch

It is very **important** to review all issues and pull requests and add the corresponding ones to the [ release milestone](https://github.com/KratosMultiphysics/Kratos/milestone/4)

@KratosMultiphysics/team-maintainers please fill the milestone until tomorrow.

**Comment by maceligueta** [comment]
I just wanted to point out that every Issue and Pull Request has the possibility to assign a _reviewer_, _asignees_, _labels_, _projects_, and **milestones**. Choosing the milestone is the proper way to tell the Release Manager that an Issue or a Pull Request is a must for a certain Release.

**Comment by pooyan-dadvand** [comment]
Thanks @maceligueta for the explanation! :+1:

**Comment by pooyan-dadvand** [comment]
# 7 days to making release branch

I assume the list of codes to be compiled is almost done. Nevertheless some applications are still missing: 
- @antonialarese MPM application is not included yet in binaries. Does it compile and prepare for the binaries in this release?
- @KratosMultiphysics/ale and @mpentek the ALE application is again missing. Please add it if is necessary to avoid a new release afterward.

Looking to the [ release milestone](https://github.com/KratosMultiphysics/Kratos/milestone/4) the most important issues are the compilation in windows:
- The #888 is done and to be merged after travis finishes all tests.
- The #849 is reproduced and I am working on it. 

Then there are several issues with **bug label** which to be tackled now.

Finally I would like to ask @KratosMultiphysics developers to give priority to the release tasks until next week

**Comment by pooyan-dadvand** [comment]
# 6 days to making release branch

Some important open issues:
- #115 for including tetgen becomes a serious problem due to the change of License in Tetgen. @KratosMultiphysics/pfem team fast reaction and support on this issue and related PR #906 is really needed.
- #635 for including pastix is more advanced and I suggest @RiccardoRossi to take care of it. 
- #849 needs to be checked by @KratosMultiphysics/dem to be merged
- #895 is still a serious problem assigned to @roigcarlo  
- #891 is implemented and to be passed by travis
- #428 is implemented and @jcotela will adopt it to trilinos application before merge.- 
- #897 almost finished to be checked by @KratosMultiphysics/structural-mechanics team.

Meanwhile:

- @antonialarese and @iiaconeta MPM application is not included yet in binaries. Does it compile and prepare for the binaries in this release?
- @playpaolo Do you want to include the ULF application?

Again I would like to ask @KratosMultiphysics developers to give priority to the release tasks in the [milestone](https://github.com/KratosMultiphysics/Kratos/milestone/4) until next week

**Comment by pooyan-dadvand** [comment]
# 5 daysto making release branch

There was a good step forward and many issues closed from milestone! Thanks all!

Important decisions:
- PFEM application will be excluded from this branch due to |#115
- MPM application will be in release binary.

Important open issues:
- #115 is still open. @KratosMultiphysics/pfem feedback for closing it is required.
- Structural Mechanic memory error in #915 and #919 and @KratosMultiphysics/structural-mechanics team is working on it
- #287 is open and assigned to @pablobecker 
- #429 is still open with a memory error to be detected

Thanks all @KratosMultiphysics developers for their effort!

**Comment by RiccardoRossi** [comment]
@pooyan-dadvand i am wondering if it would be possible to delay the release branching of about one week, say until the 7th or 8th of november. 

The point is that we still have a pretty important problem ongoing with the beam local axis (the memory errors are somewhat related to this) and this week a lot of people will be on holidays, making difficult to fix things.
It is true that we could still fix it after the branching, however it would ease our life avoiding having to patch two different branches.

**Comment by pooyan-dadvand** [comment]
In my opinion is an important feature to be included. Knowing this we have two options:
- Delay the branch for one week but still maintaining the final release time as it was.  
- Make the branch and apply the fix on both side. 

I will wait until tomorrow to see how everything evolves in milestone. Please keep this issue informed if there are some advances.

**Comment by pooyan-dadvand** [comment]
# 2 days to making release branch

Thanks all developers for their work in closing issues and problems listed in milestone. There are many important issues closed. 

Unfortunately there are still some important ones open:
- #932 is a serious blocker and its developers are on vacation this week. 
- #77 would be too deep to be corrected now and we may set it as known issue.
- #287 in enrichment cannot be addressed by @pablobecker as he is on paternal leave (Hi Anna!). Nevertheless the bug is marginal to the normal use. 
- #848  @RiccardoRossi will substitute @pablobecker.
- #635  for pastix inclusion is still open but can be added at any moment.

**Comment by jcotela** [comment]
I would also leave #872 open, since benchmarking can be done in parallel to the branch freeze.

**Comment by pooyan-dadvand** [comment]
Exactly! I have not mentioned it as it is not urgent before branch!

**Comment by josep-m-carbonell** [comment]
In the release 5.1 the SolidMechanics was included. For me it was important to be able to use the application in class in this new course. Now some of the new developments into the application and in the interface are not finished, so I  think is not convenient to do finish them fast at this moment and make some mistakes. In my case I prefer my students to download the previous version, release 5.1. 
My question here is that, would be the release 5.1 still available in the GiD repository or in some other place to download it ? Can the release 5.2 include the previous standard version offered in the release 5.1 for applications that prefer to release once or twice a year ?
Thank you in advance for you response.

@KratosMultiphysics/technical-committee   @KratosMultiphysics/team-maintainers

**Comment by pooyan-dadvand** [comment]
My answer would be yes if is technically possible. In this case I'm not sure if the solid mechanic application in 5.1 is compatible with kernel and other modules in 5.2. If is compatible I don't see any reason to forbid it.

**Comment by jginternational** [comment]
In the GiD server, the students can only download the latest version, but we can keep the 5.1 version in a server where users can download and extract into the Problemtype folder. 

I'm with  @josep-m-carbonell  , it's better not to run at the last moment again

**Comment by pooyan-dadvand** [comment]
Considering the bank holidays these days in Germany and some important open issues to be solved by TUM colleague the **release branch** will be postponed to **Tuesday (November 7th)**. 

Meanwhile I would encourage all developers to avoid pushing drastic changes to master branch.

**Comment by pooyan-dadvand** [comment]
@jginternational I also agree with @josep-m-carbonell. The problem is I don't know how to make it work..... Of course any suggestion is welcomed.

**Comment by jginternational** [comment]
@josep-m-carbonell Correction from my previous message:
GiD internet retrieve allows to download previous versions of a problemtype, so there should be **no problem** about having both versions.

The only "trouble" will be handling both versions locally, that will be impossible (both versions are kratos.gid).

**Comment by maceligueta** [comment]
I just talked to the GiD guys, and there's a partial solution to @josep-m-carbonell 's problem (which might be anyone's problem in the future). 
We could upload to the GiD server -internet retrieve- all the releases of Kratos. So the students can download older versions if required. This is possible, as it is being done by Ramseries and other codes.
This does not fix all problems, but is a partial solution until we find a better one, isn't it?

**Comment by josep-m-carbonell** [comment]
I think that keeping the previous versions somewhere is a good idea. @pooyan-dadvand @jginternational  @maceligueta . However I will try to enter in the new release if #954 is approved.  The GiDInterface is working with the changes included in it.

**Comment by roigcarlo** [comment]
@ALL I've just created the new branch for the Release-5.2. You can find it here: https://github.com/KratosMultiphysics/Kratos/tree/Release-5.2

From this moment on changes to master will not be present in the release. If you still need to commit something for the release please make a separate PR against the release branch.

Happy Release!

**Comment by pooyan-dadvand** [comment]
# 9 days to release 5.2

It is time to start finish the release. Important steps:
- Deciding the DEM application situation (to be decided by @KratosMultiphysics/dem team)
- Testing the release apps via GUI
- Creating Application cases and adding them to Examples repository

@KratosMultiphysics/team-maintainers it would be great if you add application cases as much as possible to the Examples repository

**Comment by roigcarlo** [comment]
Dear all, 
Just let you know that here you can find the first beta candidate:

https://github.com/KratosMultiphysics/Kratos/releases/tag/v5.2-Beta-1

If there is something missing / wrong, let me know.

**Comment by josep-m-carbonell** [comment]
The gid interface in the linux pack is 5.1 not the current one. It is wrong and can not be tested.

**Comment by roigcarlo** [comment]
Hi, yes GiD Interface is not ready yet. Sorry I didn't mention that, only
binaries. As soon as I have the newest version will upload it.

**Comment by pooyan-dadvand** [comment]
@josep-m-carbonell thanks for mention it. @roigcarlo try to pack it with the current repo version of GiD interface.

**Comment by roigcarlo** [comment]
@pooyan-dadvand There is no tag for Release, If you want I can pack it using the last version in master... is that ok?

**Comment by josep-m-carbonell** [comment]
Maybe you can create a branch from current master called Release-5.2 for the GiD interface. You must modify the file kratos_deffault.spd and comment appLinks that are not in the release. Thats all.

**Comment by roigcarlo** [comment]
Ok, let me ask at the GiD office, I see some branches with bugfixes and stuff for the 5.2

**Comment by roigcarlo** [comment]
@josep-m-carbonell I've updated the GiD interface to use the last version of master. This should let you run with GiD. As soon as I get a Release branch I will start using that one. I've also included a couple of missing apps.

**Comment by pooyan-dadvand** [comment]
# 8 days to release 5.2

Important notes:
DEM will use a separate binary and problemtype (like release 5.1).

The next important step would be to test the  [beta release](https://github.com/KratosMultiphysics/Kratos/releases/tag/v5.2-Beta-1) apps via GUI.  @KratosMultiphysics/structural-mechanics , @KratosMultiphysics/fluid-dynamics would you please give me some feedback about your GUI testing?

As mentioned in minutes #1064 there is a plan to add application cases (as a marketing effort) to the examples. It would be great if other colleague ( @KratosMultiphysics/optimization , @KratosMultiphysics/adjointfluid, ...) also joint this effort.

**Comment by jginternational** [comment]
@roigcarlo and @josep-m-carbonell 
Use the GUI master branch as reference for testing the released executable.
I'll prepare a beta release on Monday, which will not be pushed to GiD applications system.

**Comment by roigcarlo** [comment]
:+1:

**Comment by mpentek** [comment]
Experiences with GiD13.1.9d & Kratos 5.2 beta on Windows and Ubuntu:

@KratosMultiphysics/fluid-dynamics:
 
- In developer mode:
- cannot specify _divergence_clearance_steps_ (would it actually do a divergence clearence step? I see it for the monolithic in _navier_stokes_solver_vmsmonolithic.py_)
- if _computer_reactions = Yes_ would be nice to get it as nodal result _REACTION_ in the GiD output
- cannot specify _dynamic_tau_
- time intervals: _initial_ and _total_ are always blended out (maybe I am doing something wrong here...), so I have to define my own _custom1_ramp_up_ and _custom2_remaining_
- minimal example [cfd](https://syncandshare.lrz.de/dl/fi6zBrZTLEoGZcdNoZjysSwV/cfd.gid)

@KratosMultiphysics/structural-mechanics 

- when trying to create a plane strain example keep getting the error
_Element QuadShellThickCorotationalElement Linear (2 nodes) not available for line entities on group structure STACK: Element QuadShellThickCorotationalElement Linear (2 nodes) not available for line entities on group structure  while executing..._, tried with various elements
- time interval _initial_ and _total_ are blended out
- minimal example [csd ](https://syncandshare.lrz.de/dl/fi7FefwJLQKXNeS7CSHDLVGm/csd.gid)

@roigcarlo 
ALE for FSI -> Linux and Windows version is not consistent, meaning:
under Windows _python_solvers_wrapper_ale.py_ -> for e.g. in _ale_navier_stokes_solver_fractionalstep.py_ it will look for the FluidDynamics solver _navier_stokes_solver_fractionalstep.py_ file straight away (and this needs to be specified), whereas under Linux it is not the respective fluid solver file but the  _python_solvers_wrapper_fluid.py_ which is being called, so for Windows one needs _navier_stokes_settings["solver_type"].SetString("navier_stokes_solver_fractionalstep")_ and Linux _navier_stokes_settings["solver_type"].SetString("FractionalStep")_ so it has to be checked why the construction of the solver is different

Afterwards some fixes:
- in the ALE python scrips for the _ale_navier_stokes..._ line 19 -> needs to be changed to be consistent with the FluidDynamics App
- in the ALE python scrips for the _ale_trilinos_navier_stokes..._ line 21 -> needs to be changed to be consistent with the FluidDynamics App

- good news is that both FractStep as well as Mono seems to work
- minimal example [FSI](https://syncandshare.lrz.de/dl/fi5Q27uVn5rxEPcqn5YJTbfo/fsi)
- ale python scripts to be updated from [here ](https://syncandshare.lrz.de/dl/fiNUUJ5sVQ9iDGCrCSDbg29g/ale)

In Windows 64 the path for _runkratos_ is under _kratos.gid\exec\Kratos_
In Linux the path for _runkratos_ should be under _kratos.gid\exec\Kratos_ but is under  _kratos.gid_ so GiD throws the error
_"
/home/pmt/Software/GiD1318d/problemtypes/kratos.gid/kratos.unix.bat: line 30: /home/pmt/Software/GiD1318d/problemtypes/kratos.gid/exec/Kratos/runkratos: No such file or directory
"_

For Ubuntu14: there is an error and a core dump
For Ubuntu16: there is an error but it runs through
_"
Could not find platform independent libraries <prefix>
Could not find platform dependent libraries <exec_prefix>
Consider setting $PYTHONHOME to <prefix>[:<exec_prefix>]
"_

**Comment by roigcarlo** [comment]
@mpentek Thx for the feedback. I updated the binaries so the following issues are fixed:
* Ubuntu - 14: Should run ok now. Error was caused by a libgomp incompatibility.
* Ubuntu - 16: PYTHONHOME warning should be gone. Warning was caused by the variable not being defined in the run script.
* Ubuntu - All versions: The runkratos and rest of the libraries are now correctly located in kratos.gid/exec/Kratos.

As for the ALE scripts @rubenzorrilla is looking into it. I will make a new Beta version once this is merged.

**Comment by rubenzorrilla** [comment]
Hi @mpentek!

@jcotela and me did the next changes in the fluid dynamics strategies parameters:

- The nodal result REACTION has been added to the output settings.
- dynamic_tau has been added to the fractional step strategy settings. For both monolithic and fractional step strategies we set the default value equal to 1.0.
- Regarding the time intervals, the initial one is set to be always from 0.0 to 0.0 while the total one goes from 0.0 to "End". In case the user needs another interval combination, he needs to define two new intervals as you did (this is the intended use).
- By clicking the "book" icon in the vertical left button bar next to the interface tree, a simple cylinder test case is automatically set. Is this example what you meant by "minimal example"?

 Finally, both @jcotela and me agree that we can remove the divergence clearance from the solvers if you don't use it. What do you think?

**Comment by mpentek** [comment]
@rubenzorrilla  @jcotela Agree to all points, the last -> divergence clearance step:
We generally use a ramp-up in time to (a) be more physical and (b) numerically more stable. So it does not affect us. The thing is, if it is there in the default parameters and implemented, than it would be nice way to be able to turn it on-off from GiD. If it does not exist, than it should disappear to avoid confusion.

I only have a brief experience with using it, but it did not seem to bring much if set to on with 10-25 steps other than make the simulation longer. What is your experience with it?

Ah, no, by _minimal example_ I mean I attached one through the URL if needed :-)

**Comment by jcotela** [comment]
@mpentek In my experience, divergence clearance is not so useful. I think part of the issue is that it initializes the problem to a pure Stokes flow, which, while divergence-free, is still quite far from a moderate/high-Reynolds solution.

👍 to removing it (first from the interface, then from the application).

**Comment by rubenzorrilla** [comment]
+1 about removing it

**Comment by mpentek** [comment]
Is it planned to have a GiD release as well or to use Kratos 5.2 with GiD 13.1 or 13.1.9d?

**Comment by jginternational** [comment]
@mpentek There will be a [release of the GiD Problemtype ](https://github.com/KratosMultiphysics/GiDInterface/issues/160) compatible with GiD 13.1.9d

**Comment by mpentek** [comment]
@jginternational great news, as we were testing it with this setup :-)

**Comment by mpentek** [comment]
@KratosMultiphysics/structural-mechanics - **IMPORTANT**: please leave a short note with respect to this error (if it at all occurs for you):

when trying to create a plane strain example keep getting the error
_Element QuadShellThickCorotationalElement Linear (2 nodes) not available for line entities on group structure STACK: Element QuadShellThickCorotationalElement Linear (2 nodes) not available for line entities on group structure while executing...,_ tried with various elements

A CSD solve with an already existing example, so it might be that this bug occurs when creating a brand new example.

**Comment by philbucher** [comment]
Hi @mpentek 
I just tried, I don't get this error.
I am using GiD13.1.8d with the current Masterbranch of the Problemtype (there is no beta release yet right @jginternational )
Mate you use 13.1.9d?

**Comment by mpentek** [comment]
@philbucher Do not use the master branch, but the beta release: https://github.com/KratosMultiphysics/Kratos/releases/tag/v5.2-Beta-1
I tried it out using 13.1.8d under Linux and 13.1.9 Windows. But maybe I did something wrong.

**Comment by philbucher** [comment]
@mpentek 
I don't get the meshing error, I hope I was using the same setup now
@roigcarlo I do get a different error though, #1053 seems not to be in the release. Without this the StructuralMechanics Stuff wont run, can you please include it. Is this still possible?
@mpentek you didnt get this far thats why you didn't get this error, right?

**Comment by GuillermoCasas** [comment]
@jcotela That has been my experience too.

**Comment by mpentek** [comment]
@philbucher yes, probably I did not get that far. Did you try to create a cantilever-like structure meshed with quadrilateral THIN or THICK shell (the one we were using for the FSI example in the course, which you were preparing the mapping for) with point load on the lines?

**Comment by mpentek** [comment]
@philbucher  this is the minimal example I was using [csd ](https://syncandshare.lrz.de/dl/fi7FefwJLQKXNeS7CSHDLVGm/csd.gid)

**Comment by jginternational** [comment]
https://github.com/KratosMultiphysics/GiDInterface/releases/tag/Release-5.2.0

here it is, only the GUI files (no executables yet).

We'll have it there for testings and bugfixes (there is a branch for it) until Monday, when I'm taking it into the GiD Internet Retrieve system

**Comment by roigcarlo** [comment]
Here you have the new release. This uses @jginternational 's new GiD interface and also has #1096 as well as all fixes from the first beta.
https://github.com/KratosMultiphysics/Kratos/releases/tag/v5.2-Beta-2

Edit:
~~Win32 version is still on the oven, but as virtualy nobody uses it you can start testing!~~

**Comment by pooyan-dadvand** [comment]
Just to comment that at this moment the last beta version is:

https://github.com/KratosMultiphysics/Kratos/releases/tag/v5.2-Beta-2

**Comment by pooyan-dadvand** [comment]
@roigcarlo I just see your comment

**Comment by mpentek** [comment]
Thanks, we will have another go at it.

**Comment by philbucher** [comment]
I get the same error as @mpentek 
@jginternational any idea what might be the problem?

**Comment by mpentek** [comment]
Kratos 5.2.1beta experiences (Winddows 10 x64 and Ubuntu16 x64)

@jcotela @rubenzorrilla  **CFD**
When I select Parts -> material Air it does not seem to take the predefined values from Air in Materials

Own example
If I set the gravity value to 9.81 (or something not 0.0) I get a simulation that will eventually blow up (this bug was already present in earlier versions as well -> symptom here is getting a backflow at the outlet -> fractional step seems to be affected more/worse)

Cylinder (default) example
Works, but when gravity is non-zero it seems to be wrong

@philbucher **CSD**

Truss (default) example
Seems to work, akthough hard to visualize

Structural example still get
_Error (l685): Element QuadShellThinCorotationalElement Linear (2 nodes) not available for line entities on group structure
STACK:_
which eventually leads to _C:\PROGRA~1\GiD\GID131~1.8D\PROBLE~1\kratos.gid\exec\kratos\runkratos.exe: can't open file 'MainKratos.py': [Errno 2] No such file or directory_

@roigcarlo **FSI** 
Own example

works without a problem (->so CSD should also be ok, but I cannot seem to create a working dynamic analysis example...)

**Comment by ipouplana** [comment]
I just tried kratos-5.2.1-win-64 in the Poromechanics Application and all the examples ran flawlessly except for one in which I use the order platform.system() to determine the OS:

```
Traceback (most recent call last):
  File "MainKratos.py", line 139, in <module>
    ProjectParameters["solver_settings"]["move_mesh_flag"].GetBool())
  File "C:\Users\Ignasi\Downloads\kratos.gid\exec\Kratos\applications\PoromechanicsApplication\python_scripts\poromechanics_fracture_propagation_utility.py", line 26, in __init__
    if platform.system()=="Windows":
  File "C:\Users\Ignasi\Downloads\kratos.gid\exec\Kratos\python35.zip\platform.py", line 1097, in system
  File "C:\Users\Ignasi\Downloads\kratos.gid\exec\Kratos\python35.zip\platform.py", line 992, in uname
  File "C:\Users\Ignasi\Downloads\kratos.gid\exec\Kratos\python35.zip\platform.py", line 573, in win32_ver
  File "C:\Users\Ignasi\Downloads\kratos.gid\exec\Kratos\python35.zip\platform.py", line 507, in _get_real_winver
  File "C:\Users\Ignasi\Downloads\kratos.gid\exec\Kratos\python35.zip\ctypes\__init__.py", line 7, in <module>
ImportError: No module named '_ctypes'
KRATOS TERMINATED WITH ERROR
```

**Comment by rubenzorrilla** [comment]
@mpentek 

For the air issue, you need to refresh the material properties with the small button in the right hand side of the material name.

When gravity is non-zero, do you switch on the hydrostatic component in the outlet?

**Comment by mpentek** [comment]
@rubenzorrilla Ok, I will use the refresh, thanks for the hint. No, I did not switch on the hydrostatic component. I guess that is the issue, right?

**Comment by rubenzorrilla** [comment]
@mpentek  Might be! Give us some feedback! :wink:

**Comment by ipouplana** [comment]
Now I tried kratos-5.2.1-linux-64.tar.gz in the Poromechanics Application and all the examples ran correctly without the error of Windows:

> I just tried kratos-5.2.1-win-64 in the Poromechanics Application and all the examples ran flawlessly except for one in which I use the order platform.system() to determine the OS:
> 
> Traceback (most recent call last):
>   File "MainKratos.py", line 139, in <module>
>     ProjectParameters["solver_settings"]["move_mesh_flag"].GetBool())
>   File "C:\Users\Ignasi\Downloads\kratos.gid\exec\Kratos\applications\PoromechanicsApplication\python_scripts\poromechanics_fracture_propagation_utility.py", line 26, in __init__
>     if platform.system()=="Windows":
>   File "C:\Users\Ignasi\Downloads\kratos.gid\exec\Kratos\python35.zip\platform.py", line 1097, in system
>   File "C:\Users\Ignasi\Downloads\kratos.gid\exec\Kratos\python35.zip\platform.py", line 992, in uname
>   File "C:\Users\Ignasi\Downloads\kratos.gid\exec\Kratos\python35.zip\platform.py", line 573, in win32_ver
>   File "C:\Users\Ignasi\Downloads\kratos.gid\exec\Kratos\python35.zip\platform.py", line 507, in _get_real_winver
>   File "C:\Users\Ignasi\Downloads\kratos.gid\exec\Kratos\python35.zip\ctypes\__init__.py", line 7, in <module>
> ImportError: No module named '_ctypes'
> KRATOS TERMINATED WITH ERROR

**Comment by roigcarlo** [comment]
@ipouplana Thx for the feedback. I am looking into the win error. I identified it but so far no luck fixing it.
@mpentek Not sure if I can help with the FSI example

**Comment by mpentek** [comment]
@roigcarlo the FSI works, only if I setup a brand new CSD I have a problem, so ALE and FSI are good :-)

**Comment by mpentek** [comment]
@roigcarlo 
Running Kratos 5.2.1 beta through GiD and terminal in Ubuntu 16 works
Running Kratos 5.2.1 beta through GiD works, but running it via terminal in Ubuntu 14 produces a core dump with the similar errors:
"
Could not find platform independent libraries <prefix>
Could not find platform dependent libraries <exec_prefix>
Consider setting $PYTHONHOME to <prefix>[:<exec_prefix>]
Python error: Py_Initialize: Unable to get the locale encoding
Error: No module nemad 'encodings'
"

Are we setting up/using something under Ubuntu14 incorrectly or is this really due to a difference between 14 and 16?

**Comment by roigcarlo** [comment]
Hi @mpentek, in order for terminal version to work in 14.04 you must 
export PYTHONHOME=/path/to/runkratos

That should make it work, but I don't like that solution... (It works under GiD because we do that)

I'm gonna try to embed it inside the binary but I am not sure if it can be done in 14.04.
I honestly don't know why is there a problem in 14 and not in 16 :S

**Comment by mpentek** [comment]
@roigcarlo The problem is in 14 and not 16! Ok, we will define the PYTHONHOME in 14 as we anyway use a setup.sh file for what is necessary. The solution should be fine with us, will test it out.

**Comment by roigcarlo** [comment]
up sorry, misplaced the version numbers :P

**Comment by mpentek** [comment]
@jginternational  could you give some feedback related to setting up and running a Structural Mechanics example from GiD?

Structural example still get
Error (l685): Element QuadShellThinCorotationalElement Linear (2 nodes) not available for line entities on group structure
STACK:
which eventually leads to C:\PROGRA1\GiD\GID1311.8D\PROBLE~1\kratos.gid\exec\kratos\runkratos.exe: can't open file 'MainKratos.py': [Errno 2] No such file or directory

It seems that the _runkratos_ can solve it, but there is a problem from the GiD side

**Comment by philbucher** [comment]
FYI:
 @KratosMultiphysics/structural-mechanics started to add [examples](https://github.com/KratosMultiphysics/Examples/tree/master/structural_mechanics) in the new Examples repo

**Comment by jginternational** [comment]
@mpentek I'm out of the office but, the message:

> Element QuadShellThinCorotationalElement Linear (2 nodes) not available for line entities on group structure

means that you are applying a Thin corotational quadrilateral shell to a 2 noded line. This element should be applied to 4 noded surface element (Quadrilateral).

One of our pending issues is to create a "[nice error window](/KratosMultiphysics/GiDInterface/issues/146)"

**Comment by pooyan-dadvand** [comment]
> FYI:
> @KratosMultiphysics/structural-mechanics started to add examples in the new Examples repo

Thanks @KratosMultiphysics/structural-mechanics ! Nevertheless for some reason, the images are not loading in the [github page](https://kratosmultiphysics.github.io/Examples/structural_mechanics/validation/beam_roll_up/)

**Comment by pooyan-dadvand** [comment]
Indeed the new benchmarks are really nice!

**Comment by philbucher** [comment]
@KlausBSautter I think you have to move the pictures also to the Examples repo for them to show up

**Comment by mpentek** [comment]
@jginternational it would be quite important to solve this issue, see my detailed experience as follows with sample files hyperlinked. I think there is some problem with the model creation from the GiD side. for an FSI case I have a previous shell model and works well for solving. however, anything surface-like I currently try to create does not. line (beam) and volume (solid) type elements work fine now as well. I am quite confident that the model is created correctly. 

the error message definitely says that some entities are trying to be assigned/combined with conditions that are incompatible. in case you are out of office for a longer time is there a chance that this issue is addressed by someone else?

@KratosMultiphysics/structural-mechanics maybe you have some similar/other experience

**Samples which DO NOT seem to work:**
**thin triangular shell**
-> [static](https://syncandshare.lrz.de/dl/fiBAPEvKNDkWTnMHjZ99P5dZ/csd_shell_thin_triangular_static.gid), [dynamic](https://syncandshare.lrz.de/dl/fiP5zynoMN7gzvzkiynzxxG3/csd_shell_thin_triangular_dynamic.gid)

_Error (l685): You have not assigned a proper entity to group structure
STACK:
You have not assigned a proper entity to group structure
    while executing_

**thick quadrilateral shell**
-> [static](https://syncandshare.lrz.de/dl/fiN4YWCrtJ7Nt5CViJzUAGLK/csd_shell_thick_quadrilateral_static.gid), [dynamic](https://syncandshare.lrz.de/dl/fi54NxtC3fJ5sCkE99ayH7Pm/csd_shell_thick_quadrilateral_dynamic.gid)

_Error (l685): You have not assigned a proper entity to group structure
STACK:
You have not assigned a proper entity to group structure
    while executing_

**thin corotational quadrilateral shell**
->[ static](https://syncandshare.lrz.de/dl/fi4c5WP2e8AaZTyZRYzMNf2e/csd_shell_thin_corotational_quadrilateral_static.gid), [dynamic](https://syncandshare.lrz.de/dl/fiTf1TdWDHhzS5YRNCdEXyg5/csd_shell_thin_corotational_quadrilateral_dynamic.gid)

_Error (l685): Element QuadShellThinCorotationalElement Linear (2 nodes) not available for line entities on group structure
STACK:
Element QuadShellThinCorotationalElement Linear (2 nodes) not available for line entities on group structure
    while executing_

 
 
 
 
**Samples which DO work:**
- [solid small displacement tetrahedra static](https://syncandshare.lrz.de/dl/fiJdyvJW4SdXmGH6Cz8D7n1f/csd_solid_small_displacement_tetrahedra_static.gid)
- [solid total lagrangian hexahedra dynamic](https://syncandshare.lrz.de/dl/fiEPkAqc9G56Vj7UJeTpJdBR/csd_solid_total_lagrangian_hexahedra_dynamic.gid)
- [beam large displacement static](https://syncandshare.lrz.de/dl/fi89mA79ETLeBqbMvufRYNRS/csd_large_displacement_beam_static.gid)

**Comment by jginternational** [comment]
By the end of tomorrow (05/12/17) I should have this fixed @mpentek

**Comment by mpentek** [comment]
@jginternational great, looking forward to it

@roigcarlo we added PYTHONHOME for Ubuntu14 (as it can be seen in the unix_realease.bat, but still get when trying to run it through the terminal:

_Fatal Python error: Py_Initialize: Unable to get locale encoding
ImportError: No module named 'eoncodings'

Current thread...
Aborted (core dumped)_


I was reading that PYTHONPATH might be wrong and conflicting. We generally have an alias in the .bashrc which calls the following lines (possible syntax issues...):


_export PYTHONPATH=$PYTHONPATH:/home/kodakkal/Software/gid13.1.8d-x64/problemtypes/kratos.gid/exec/Kratos
export PYTHONHOME=/home/kodakkal/Software/gid13.1.8d-x64/problemtypes/kratos.gid/exec/Kratos
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/home/kodakkal/Software/gid13.1.8d-x64/problemtypes/kratos.gid/exec/Kratos/libs
export PATH=$PATH:/home/kodakkal/Software/gid13.1.8d-x64/problemtypes/kratos.gid/exec/Kratos

runkratos() {
 /home/kodakkal/Software/gid13.1.8d-x64/problemtypes/kratos.gid/exec/Kratos/runkratos "$@"
}_

**Comment by jginternational** [comment]
@mpentek Check the issue https://github.com/KratosMultiphysics/GiDInterface/issues/210

**Comment by pooyan-dadvand** [comment]
@roigcarlo Have you addressed the @mpentek issue in your release?

**Comment by roigcarlo** [comment]
@pooyan-dadvand @mpentek 
The linux release is now a bit different and should not give problems, but if you want to use kratos without using GiD you have to set the ENV variables correctly. To my knowledge the vars needed are:

- PYTHONPATH: should point to both kratos folder and python34.zip
- PYTHONHOME: should point to kratos folder
- LD_LIBRARY_PATH: should point to kratos/libs

In conclusion, same thing as if you were using the compiled version but you also have to add PYTHONHOME. This needs to be this way because we cannot asume a native python exists in the system...

If it is easier i can set up a minimal .sh to guide the users.

**Comment by mpentek** [comment]
@roigcarlo yes, we intend to use it with GiD as well as from terminal with students for teaching and similar activities, so it should be robust. Also, our internal OS are Win7 or Ubuntu14 so it is important that it works. I guess that most new users will appreciate these aspects.

Our sample .sh includes

_export PYTHONPATH=$PYTHONPATH:/home/kodakkal/Software/gid13.1.8d-x64/problemtypes/kratos.gid/exec/Kratos
export PYTHONHOME=/home/kodakkal/Software/gid13.1.8d-x64/problemtypes/kratos.gid/exec/Kratos
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/home/kodakkal/Software/gid13.1.8d-x64/problemtypes/kratos.gid/exec/Kratos/libs
export PATH=$PATH:/home/kodakkal/Software/gid13.1.8d-x64/problemtypes/kratos.gid/exec/Kratos

runkratos() {
/home/kodakkal/Software/gid13.1.8d-x64/problemtypes/kratos.gid/exec/Kratos/runkratos "$@"
}_

To be able to call the _runkratos_. Nonetheless, it seems that we are not setting the PYTHONHOME correctly in it. Ill try it out once more on Ubuntu14 again.

**Comment by mpentek** [comment]
@roigcarlo it works with this SH file for Ubuntu14, for Ubuntu16 it was working anyway, so I guess it still does :-)
[kratosPreSetup521.txt](https://github.com/KratosMultiphysics/Kratos/files/1535315/kratosPreSetup521.txt)

**Comment by mpentek** [comment]
@pooyan-dadvand thanks for the help and pushing with the release. Will it be released in the next couple of days? We would anyway use it for CFD in teaching from next week and move on to CSD and FSI in January. Will post the experiences, just need to know whether it will be available with internet retrieve inside GiD or some manual copy and setup. 

Here are my notes so far:

**Running and setup on various OSs**
@roigcarlo 

It seems that it will run with GiD as well as from terminal for Win 7 & 10 +Ubuntu 14 & 16. Out of curiosity: do you think there is a relatively easy way to make it run for macOS or one will have to stick with the developer for that?

**CFD**
@jcotela @rubenzorrilla 

It is very nice to have REACTION from the GUI and having removed divergence clearance step to avoid confusion.
We did not manage to thouroughly test the gravity settings and adding the hydrostatic contribution to the pressure outlet condition. The results with gravity on (and hydrostatic on or off) seem strange, but for airflow I am not concerned and will use it withour gravity for now.

**CSD**
Some examples seem to work well, others seem to have some bug, which might rather be related to GiD during the model creation as I can run exsinting older examples with runkratos.

**FSI**
Existing example works withouth a problem, so ALE seems to be ok.

**GiD**
@jginternational 

Thanks for the help and tips for now. I think the issue with some 2D-type structural elements is not totally cleared. But more on this in #210. Today we also encountered a bug when meshing, so I will either post it there or in a different issue. I think neither of these 2 is very typical and quite case-dependent, so we will see.

**Comment by roigcarlo** [comment]
@mpentek Sure, it is planed, we just didn't have time. We have a issue with python that we need to solve first but it is almost there

**Comment by pooyan-dadvand** [comment]
@mpentek our plan is to push it today to the GiD internet retrieve. Actually I only miss the whats new from the teams to upload it. It would be great if you use it in your courses and give us your valuable feedback!

**Comment by pooyan-dadvand** [comment]
@KratosMultiphysics/structural-mechanics, @KratosMultiphysics/fluid-dynamics and @rubenzorrilla would you please add the corresponding what's new section of [release 5.2](https://github.com/KratosMultiphysics/Kratos/releases)

**Comment by rubenzorrilla** [comment]
@pooyan-dadvand, @RiccardoRossi already did it. If something comes to my mind I will add it.

**Comment by pooyan-dadvand** [comment]
I am pleased to announce the [**release 5.2.0**](https://github.com/KratosMultiphysics/Kratos/releases/tag/v5.2.0)

It is also uploaded to GiD internet retrieve, so the GiD users can directly download it from there. 

Thanks to all @KratosMultiphysics developers for their support

**Comment by mpentek** [comment]
@roigcarlo 

Early experiences:
Windows10: GiD and Kratos work, Kratos from shell works
Ubuntu14: GiD and Kratos work, Kratos from terminal works
Ubuntu16: GiD and Kratos work, Kratos from terminal does not -> error about not finding libpython although it is there:
"
/home/pmt/Software/KratosPrecompiled520/kratos.gid/exec/Kratos/runkratos: error while loading shared libraries: libpython3.4m.so.1.0: cannot open shared object file: No such file or directory
"

See the setup file (note, the same exact one works on Ubuntu14, and Kratos5.2.1beta was also working on Ubuntu16 with a similar script):
[kratosPre520Setup txt](https://github.com/KratosMultiphysics/Kratos/files/1542408/kratosPre520Setup.txt)

**Comment by roigcarlo** [comment]
@mpentek That's odd. If @KratosMultiphysics/technical-committee agrees, I will pack next version with an extra "launchKratos.sh" that you will simply call like:

```bash
launchKratos.sh Script.py
```

and already has all the paths setted.

In the meantime, just add:

```bash
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/home/pmt/Software/KratosPrecompiled520/kratos.gid/exec/Kratos
```

to the file you sent me and it should work

**Comment by mpentek** [comment]
@roigcarlo 
Thanks, works like a charm. It seems very odd indeed...

**Comment by pooyan-dadvand** [comment]
Thanks everybody! I just closing the issue to open a new one for release 5.3! :scream: :smile:

**Comment by xuhenry** [comment]
@mpentek Hello, Will you tell me where can I find the example files of CSD? Thanks  a lot!
Henry

**Comment by pooyan-dadvand** [comment]
I would suggest you to check the latest [release](https://github.com/KratosMultiphysics/Kratos/releases/tag/7.0) and the latest [tutorial](https://github.com/KratosMultiphysics/Kratos/wiki/Running-an-example-from-GiD) with the CSD sample described in https://github.com/KratosMultiphysics/Kratos/wiki/Data-management

**Comment by pooyan-dadvand** [comment]
@KratosMultiphysics/structural-mechanics
